File: adapters/python/testdata/mypackage/service.py
from abc import ABC, abstractmethod


class Greeter(ABC):
    """Abstract greeter interface."""

    @abstractmethod
    def greet(self, name: str) -> str: ...


class HelloService(Greeter):
    """Concrete implementation of Greeter."""

    def __init__(self, prefix: str = "Hello") -> None:
        self.prefix = prefix

    def greet(self, name: str) -> str:
        return f"{self.prefix}, {name}!"

    def farewell(self, name: str) -> str:
        return f"Goodbye, {name}!"


def create_service(prefix: str = "Hello") -> HelloService:
    """Factory function."""
    return HelloService(prefix)

File: adapters/rust/testdata/src/service.rs
pub const DEFAULT_GREETING: &str = "Hello";

pub trait Greeter {
    fn greet(&self) -> String;
    fn target(&self) -> &str;
}

pub struct EnglishGreeter {
    name: String,
}

impl EnglishGreeter {
    pub fn new(name: &str) -> Self {
        EnglishGreeter {
            name: name.to_string(),
        }
    }
}

impl Greeter for EnglishGreeter {
    fn greet(&self) -> String {
        format!("{}, {}!", DEFAULT_GREETING, self.name)
    }

    fn target(&self) -> &str {
        &self.name
    }
}
